collector/system_collector.go

```diff
@@ -62,63 +72,86 @@ func (sc *systemCollector) Describe(ch chan<- *prometheus.Desc) {
 
 // Collect implements the prometheus.Collector interface.
 func (sc *systemCollector) Collect(ch chan<- prometheus.Metric) {
-	sc.collectClusterStatusMetric(ch)
-	sc.collectClusterNodeMetric(ch)
-	sc.collectClusterServiceMetric(ch)
+	statusMetrics := sc.collectClusterStatusMetrics()
+	for _, sm := range statusMetrics {
+		ch <- prometheus.MustNewConstMetric(sc.clusterStatus, prometheus.GaugeValue, sm.Status)
+	}
+
+	nodeMetrics := sc.collectClusterNodeMetrics()
+	for _, nm := range nodeMetrics {
+		ch <- prometheus.MustNewConstMetric(sc.clusterNodeStatus, prometheus.GaugeValue, nm.Status, nm.IPAddress, nm.Type)
+	}
+
+	serviceMetrics := sc.collectClusterServiceMetrics()
+	for _, svm := range serviceMetrics {
+		ch <- prometheus.MustNewConstMetric(sc.applianceManagementServiceStatus, prometheus.GaugeValue, svm.Status)
+	}
 }
 
-func (sc *systemCollector) collectClusterStatusMetric(ch chan<- prometheus.Metric) {
+func (sc *systemCollector) collectClusterStatusMetrics() (systemStatusMetrics []systemStatusMetric) {
 	clusterStatus, err := sc.systemClient.ReadClusterStatus()
 	if err != nil {
 		level.Error(sc.logger).Log("msg", "Unable to collect cluster status")
 		return
 	}
-
+	systemStatusMetric := systemStatusMetric{
+		Status: 0.0,
+	}
 	if strings.ToUpper(clusterStatus.ControlClusterStatus.Status) == "STABLE" &&
```

Comment:
Just wondering, is this "STABLE" equal with all Cluster Node status "CONNECTED"?